apps/end-tenancy/models/upload.js

```diff
@@ -2,69 +2,77 @@
-      };
+      const formData = new FormData();
+      formData.append('document', this.get('data'), {
+        filename: this.get('name'),
+        contentType: this.get('mimetype')
+      });
+      reqConf.data = formData;
       reqConf.method = 'POST';
+      reqConf.headers = {
+        ...formData.getHeaders()
+      };
       debug('SAVE PDF DATA', reqConf);
-      this.request(reqConf, (err, data) => {
-        if (err) {
-          return reject(err);
-        }
-        debug('RESPONSE FROM FILE VAULT SAVE', err, data);
-        return resolve(data);
+      const responseData = await new Promise((resolve, reject) => {
+        this.request(reqConf, (err, data) => {
+          if (err) {
+            return reject(err);
+          }
+          debug('RESPONSE FROM FILE VAULT SAVE', err, data);
+          return resolve(data);
+        });
       });
-    });
-  }
 
-  auth() {
-    if (!config.keycloak.tokenUrl) {
-      // eslint-disable-next-line no-console
-      console.error('keycloak token url is not defined');
-      return Promise.resolve({
-        bearer: 'abc123'
-      });
+      await this.set({ url: responseData.url });
+      await this.unset('data');
+
+      return responseData;
+    } catch (err) {
+      console.error('Error in save method ', err);
+      throw err;
     }
-    const tokenReq = {
-      url: config.keycloak.tokenUrl,
-      form: {
-        username: config.keycloak.username,
-        password: config.keycloak.password,
-        grant_type: 'password',
-        client_id: config.keycloak.clientId,
-        client_secret: config.keycloak.clientSecret
-      },
-      method: 'POST'
-    };
+  }
 
-    return new Promise((resolve, reject) => {
+  async auth() {
+    /* eslint-disable no-console */
+    try {
+      if (!config.keycloak.tokenUrl) {
+        console.error('keycloak token url is not defined');
+        return {
+          bearer: 'abc123'
+        };
+      }
+      const tokenReq = {
+        url: config.keycloak.tokenUrl,
+        headers: { 'content-type': 'application/x-www-form-urlencoded' },
+        data: {
+          username: config.keycloak.username,
+          password: config.keycloak.password,
+          grant_type: 'password',
+          client_id: config.keycloak.clientId,
+          client_secret: config.keycloak.clientSecret
+        },
+        method: 'POST'
+      };
       debug('REQUEST DATA', tokenReq);
-      this._request(tokenReq, (err, res) => {
-        debug('RESPONSE FROM FILE VAULT', err, res);
-        if (err) {
-          return reject(err);
-        }
-        return resolve({
-          bearer: JSON.parse(res.body).access_token
-        });
-      });
-    });
+      const response = await this._request(tokenReq);
+      debug('RESPONSE FROM FILE VAULT', response);
+      return { bearer: response.data.access_token };
+    } catch (err) {
+      console.error(`Error in auth method: ${err.response.data.error} - ${err.response.data.error_description}`);
```

Comment:
```suggestion
      console.error(`Error in auth method: ${err.response?.data?.error} - ${err.response?.data?.error_description}`);
```
In the previous code, if err.response or err.response.data was undefined, attempting to access err.response.data.error would throw an extra error, potentially hiding the original issue.

Adding optional chaining ensures that `console.error` won't fail because of undefined properties.  They'll just show as undefined and instead of throwing another error

hope that makes sense?

Reply:
Yes, that makes sense thanks 👍🏾 that way we don't run into those pesky 'can't find property of undefined' errors. I have implemented this now